tao_compiler/mlir/disc/transforms/lhlo_fusion.cc

```diff
@@ -454,7 +474,16 @@ class FusionPlanner {
     // want to make sure the normal fusion pattern (fusion cross def-use) take
     // first. Without this design, some horizontal fusion may break following
     // normal fusion pattern.
-    if (useHorizontalFusion()) {
+    bool enable_horizontal_fusion = useHorizontalFusion();
```

Comment:
maybe just try to introduce a virtual function for FusionStrategy class instead of such hack.

Reply:
`PlacementAwareFusionStrategy ` is also a `FusionStrategy ` actually. Thus it can override the virtual function as well.
For example, we can add a virtual function `bool supportHorizontalFusion()` and ``PlacementAwareFusionStrategy` returns true only when all its sub-strategies `supportHorizontalFusion()` ?

Reply:
> `PlacementAwareFusionStrategy ` is also a `FusionStrategy ` actually. Thus it can override the virtual function as well. For example, we can add a virtual function `bool supportHorizontalFusion()` and `PlacementAwareFusionStrategy` returns true only when all its sub-strategies `supportHorizontalFusion()` ?

This may work. I will try in another PR.